src/main/java/com/google/devtools/build/lib/packages/AttributeFormatter.java

```diff
@@ -347,7 +348,7 @@ private AttributeBuilderAdapter(
 
     @Override
     public void addStringListValue(String s) {
-      attributeBuilder.addStringListValue(s);
+      attributeBuilder.addStringListValue(internalToUnicode(s));
```

Comment:
I think this conversion is incorrect, since all callsites already perform the conversion to Unicode (also for the other three below)?

I will add some integration tests in bazel_query_test.sh.

Reply:
I think you are right, but I would personally drop the `internalToUnicode` calls from the callsites rather than this function as `addStringListValue` etc. are the actual sinks for valid Unicode. I don't have a strong opinion though, feel free to do what you think is easier to understand

Thanks for taking care of the tests!

Reply:
While I generally agree with the principle of putting the conversion as close to the sink as possible, in this case it would cause `writeAttributeValueToBuilder` to end up with mixed application/non-application of `internalToUnicode`, as not every proto field is set through the `AttributeValueBuilderAdapter` interface. That seems less readable to me, because it's harder to convince oneself that all fields have been covered. (It might be possible to do better with additional indirection, but I'd rather not do that now.)